modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/search/spi/model/index/contributor/DLFileEntryModelDocumentContributor.java

```diff
@@ -255,6 +250,45 @@ private String _extractDDMContent(
 		return sb.toString();
 	}
 
+	private String _extractText(DLFileEntry dlFileEntry)
+		throws IOException, PortalException {
+
+		if (DLStoreUtil.hasFile(
+				dlFileEntry.getCompanyId(), dlFileEntry.getDataRepositoryId(),
+				dlFileEntry.getName(), _getIndexVersionLabel(dlFileEntry))) {
+
+			return StreamUtil.toString(
+				DLStoreUtil.getFileAsStream(
+					dlFileEntry.getCompanyId(),
+					dlFileEntry.getDataRepositoryId(), dlFileEntry.getName(),
+					_getIndexVersionLabel(dlFileEntry)));
+		}
+
+		InputStream inputStream = _getInputStream(dlFileEntry);
+
+		if (inputStream == null) {
+			return null;
+		}
```

Comment:
Is it possible a scenario where we have a null `inputStream` but we do have the file in the DLStore? In that case, would it be OK to return what we have in the DLStore?

Reply:
If there's a file, the inputStream will never be null. If the file is empty, then the resulting text will be the empty string (which is perfectly fine).

Reply:
👍